src/enzo/ExternalBoundary_SetGalaxySimulationBoundary.C

```diff
@@ -60,6 +60,20 @@ int ExternalBoundary::SetGalaxySimulationBoundary(FLOAT time)
     ENZO_FAIL("Error in IdentifyPhysicalQuantities.\n");
   }
 
+  int DeNum = FindField(ElectronDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HINum = FindField(HIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HIINum = FindField(HIIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HeINum = FindField(HeIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HeIINum = FindField(HeIIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HeIIINum = FindField(HeIIIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HMNum = FindField(HMDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int H2INum = FindField(H2IDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int H2IINum = FindField(H2IIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int DINum = FindField(DIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int DIINum = FindField(DIIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int HDINum = FindField(HDIDensity,BoundaryFieldType,NumberOfBaryonFields);
+  int MultiSpecies = (HINum == -1) ? 0 : 1;
```

Comment:
I don't believe setting `MultiSpecies` in this way will allow for `MultiSpecies > 1` (line 332) to ever occur. Can `MultiSpecies` not just be set from the parameter?